### ISSUE TITLE
THE QUADRUPLETS — RN-8M sibling-classroom cohort

### ISSUE THREAD

**Issue by bluegrouse92-lgtm** [N-A-pixel/Ruse-Public#10]
# THE QUADRUPLETS — RN-8M sibling-classroom cohort

On 23 September 2026 Ruse created a controlled cohort of **four independent RN-8M siblings**:

- RN-8M-Q1
- RN-8M-Q2
- RN-8M-Q3
- RN-8M-Q4

Each model contains **8,523,008 parameters** and uses the same architecture:

```text
layers:            6
hidden size:       256
attention heads:   4
KV heads:          2
FFN:               704
context:           1024
vocabulary:        16,000
```

The siblings have independent deterministic identities/seeds but deliberately share architecture and classroom conditions. They are **not merged, averaged, or allowed to share weights**.

## Why the quadruplets exist

This is not simply four reruns of the same small model. The cohort is an experiment in controlled developmental variation and peer exposure.

The question is whether four small models raised under the same lessons but with independent weight trajectories develop meaningfully different strengths, failure modes or eventual specialist roles — and whether bounded exposure to frozen sibling outputs changes later learning without destroying individual lineage.

## Sibling-classroom protocol

```text
same frozen lesson
→ each sibling trains independently
→ class barrier
→ each sibling answers frozen probes independently
→ raw answers frozen before peer exposure
→ shared sibling-review artifact constructed
→ each sibling receives deterministic peer-review material
→ transfer probes measure what changed
→ individual + cohort ledgers preserved
```

Peer information is exchanged as **frozen educational material**, never as weights or hidden state. Each sibling remains an independent model lineage.

## First protocol test

Historical metadata originally reported:

- curriculum: **219 tokens**
- sibling-review material: **1,767 tokens**
- total exposure per sibling: **1,986 tokens**

Later audit showed that labels were shifted twice by the causal-LM loss path. With sequence length 64, only 63 target positions contributed per optimizer step. Each sibling made two real optimizer updates, so the verified historical optimized exposure is **126 targets per sibling**.

The 1,986 figure remains preserved as historical fixture metadata rather than silently rewritten.

Known cohort artifacts:

```text
cohort SHA-256:
baf616fa95cbb752794cc9d4019676b436cca95ec8741dffbf4ddaf68374fc5e

class SHA-256:
97ee431ec8a392c9a8540f546a2e26bed9964b496d50dc613acaa542e9a075be

sibling-review SHA-256:
7da70c80906f460a06363f723b6bcd88d...
```

## Experimental rules

1. Same lesson means the same frozen curriculum artifact.
2. All four must reach the barrier before sibling review begins.
3. Raw answers are frozen before peer exposure.
4. Peer-review material is deterministic and attributable.
5. No model receives another sibling's weights, optimizer state, activations or hidden state.
6. Transfer is measured only after peer exposure.
7. Individual developmental ledgers remain separate.
8. A cohort ledger records shared classroom events.
9. Later roles are earned by measured capability, not assigned from sibling identity.

## 24 September 2026 — real 400K sibling classroom

The quadruplet experiment moved from the small synthetic protocol fixture into a **real bounded foundation-education run on the Steam Deck**.

Each sibling was authorized for exactly **400,000 new optimized target exposures**:

```text
common base education:             395,904
sibling review: 4 × 1,024 =          4,096
                                   -------
new optimized targets each:        400,000
verified prior optimized targets:      126
projected lifetime optimized age:  400,126
```

Frozen common base stream SHA-256:
`f1aa993e6092d80609dd3ba35241f14c718523becb315a9fa913629456263a05`

The runner uses explicit logits/label alignment and accounts for 128 actual loss-contributing targets per full training step, avoiding the old fixture's double-shift defect.

### Same cohort, independent model states

The Deck may train round-robin while cohort barriers prevent any sibling from crossing into a later phase while another remains behind.

At each of four barriers all four reach the same educational boundary, are checkpointed and independently reloaded/verified, answer the same frozen probes, have output/repetition/formatting measurements frozen, and then receive a deterministic 1,024-target review block built only from the **other three siblings**.

Exact source ranges record where each sibling's material occurs, and no model sees a future snapshot.

### Checkpoint discipline

Checkpoints preserve independent model/optimizer/scheduler/RNG/cursor/token state. Promotion includes fresh reload, finite-model verification, tensor-for-tensor state-dict equality, content hashing, atomic rename and directory fsync. Cohort progress advances only through verified synchronization states.

## Intended future — small specialists

These 8M models are candidate bounded specialist runners, not miniature replacements for a larger language model. The common classroom deliberately assigns no roles in advance. Divergence is evidence.

After common education, all four general checkpoints are frozen and subjected to identical no-update specialist-discovery probes. Possible task families include routing/classification, extraction/structuring, formatting, short transformations, technical micro-tasks, intent detection, constrained generation and error detection, but roles must follow measured behaviour rather than model number.

If specialist training is later justified, it should create separately versioned descendants while preserving the general graduates.

```text
COMMON EDUCATION
→ CONTROLLED PEER EXPOSURE
→ FREEZE FOUR GENERAL GRADUATES
→ CHARACTERISE DIFFERENCES
→ IDENTIFY CANDIDATE ROLES
→ separately authorize specialist education
→ qualify specialist descendants
```

## 24 September 2026 — canonical +5M classroom launched

The four siblings later moved beyond the completed 400K classroom.

Canonical 400K parents:

| Sibling | 400K parent SHA-256 | Starting lifetime age |
|---|---|---:|
| Q1 | `3c3e2aa76fb0eba3fa13f4cc3174ae89833d77b080a5ee9c1b848791e979bdee` | 400,126 |
| Q2 | `14fea19adeef2963d40359f0a4912b8a9c90ebeef61e593f32de7ddaa027584a` | 400,126 |
| Q3 | `7f231012786f1e4305779d106100a5355b5eabfbcbe803bfd4495eb78837f33c` | 400,126 |
| Q4 | `f3fdc9718ac412f6bdb4cacced6b238e863a3a123d45f9054bde23f9c043b5a0` | 400,126 |

Each sibling receives exactly **5,000,000 new optimized target exposures** across five synchronized 1,000,000-target intervals.

- final planned lifetime age: **5,400,126**
- per interval: **998,976 common-base + 1,024 sibling-review**
- sibling review split: 341 / 341 / 342 across the other three
- baseline snapshots at age 400,126 feed interval 1
- each completed barrier feeds only the next interval
- final snapshots are evaluation evidence only
- cohort cannot advance past a barrier until all four verify
- no specialization assignment during the run

Per sibling the full stage is:

`39,060 × 128 + 5 × 64 = 5,000,000`

= **39,065 optimizer updates**.

Base stream SHA-256:
`536de036d5e1684a73b2b368d16aa5235ae3384e1b11ac4a3fb22b0746310573`

Cohort plan SHA-256:
`d186ab931916b4d4808dd7be80a0782d7caffcc94ae504fc80feca523301a093`

Recipe SHA-256:
`098ebb72b1e6bb66231e0aeaddb6245309140c213afd243b1af091459c2c1ab8`

No-update baseline losses at launch:

| Sibling | Validation | Holdout |
|---|---:|---:|
| Q1 | 6.085013 | 6.464804 |
| Q2 | 6.111329 | 6.490273 |
| Q3 | 6.071760 | 6.450245 |
| Q4 | 6.082135 | 6.478477 |

A single-writer lock was added after the earlier duplicate Q4 invocation. Qualification verified refusal of concurrent execution, explicit ledgered recovery, checkpoint/reload/resume, exact target accounting, finite loss/gradient/parameter checks and the partial-target remainder path.

**Status: synchronized +5M classroom launched with independent states and barriered peer-review protocol.**

**Comment by bluegrouse92-lgtm** [comment]
## 24 September 2026 — canonical +5M classroom launched

The four RN-8M siblings moved beyond the completed 400K classroom.

Canonical parents all start at lifetime age **400,126**. Each sibling receives exactly **5,000,000 new optimized target exposures** across five synchronized +1M intervals, ending at **5,400,126** if completed canonically.

Each interval contains **998,976 common-base + 1,024 sibling-review targets**. Review uses frozen outputs from the other three siblings only and is strictly causal across barriers.

Per sibling, the full stage is **39,065 optimizer updates / 5,000,000 loss-contributing targets**. Single-writer locking, crash recovery, checkpoint/reload/resume, finite-math checks and exact remainder handling were qualified before launch.

Canonical execution has started on the Steam Deck. No specialization assignment is made during the run.
